label-application-rule-config-service-impl/src/test/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImplTest.java

```diff
@@ -44,20 +43,26 @@
 import org.junit.jupiter.api.Test;
 
 public class LabelApplicationRuleConfigServiceImplTest {
+  private static final String SYSTEM_LABEL_APPLICATION_RULE_STR =
```

Comment:
This test is mocking the config, so mock it (i.e. return a prebuilt rule) rather than do the real parse logic and put it behind a mock. Not only is that not the responsibility of this class/test, but if a change were added to the rule structure this test would miss it silently.

Reply:
Done. Hopefully last review comment to address ;-)